### ISSUE TITLE
Loss function in discrete case

### ISSUE THREAD

**Issue by CarolaDoerr** [facebookresearch/nevergrad#830]
I am having a look at the detailed loss values in the discrete case (sequential). Two things: 
(1) for jump, some loss values are negative, whereas it would be easy to set jump as min n-jump(x) to ensure loss>0
[I think positive values, or at least easily to interpret loss values are quite crucial to understand what is going on - ATM I do not know what a -2 corresponds to (could probably check the code, but this is uncomfortable, and the fix would be easy to implement)]

(2) For OneMax, even in dimension 5000 and a small budget of 50 evals, a loss of 0 is reported for DiscreteOnePlusOne (and many other solvers) Since all OneMax values are non-negative, I believe that loss = n - OM(x). But a loss of 0 is not very credible, so I would appreciate if someone could check what is going on here. 

[this is the  first time I am "reporting an issue", so please bear with me if this form is not correctly filled]

**Comment by teytaud** [comment]
Maybe you can copy the correspond lines here ?

**Comment by CarolaDoerr** [comment]
Hm, this was from the csv files that you have given to me

**Comment by teytaud** [comment]
Starting to investigate the discrete case, which clearly did not have the attention it deserved for the moment :-)  i observe errors such as "AttributeError" for some discrete functions:

./instrum_discrete/27401280_151_0_log.out:#json: {"loss": 4172, "elapsed_budget": 1718, "elapsed_time": 259082.52260422707, "error": "AttributeError", "pseudot
ime": 171.0, "seed": -1, "function_class": "ExperimentFunction", "name": "onemax", "arity": 7, "instrum_str": "Softmax", "parametrization": "Choice(choices=Tup
le(0,1,2,3,4,5,6),weights=Array{(5000,7)})", "dimension": 35000, "batch_mode": true, "num_workers": 10, "budget": 5000, "optimizer_name": "MetaTuneRecentering"
}

**Comment by teytaud** [comment]
(for the record, other errors found are for specific algorithms: MetaModel sometimes crashes with ValueError, and SQP sometimes crashes by overflow --- both in noisy functions, I'll investigate this at least for MetaModel)

**Comment by teytaud** [comment]
(also some overflow errors for various algorithms in the very parallel case, e.g. yaparabbob, or very high-dimensional, e.g. yahdbbob --- maybe reducing the arxiv in some cases would fix various things simultaneously --- but this is unrelated to the discrete case, and these overflow errors are due to some specific optimization algorithms, not due to the framework)

**Comment by teytaud** [comment]
Sounds like this is solved.
